### ISSUE TITLE
Lab 5 level 5 jshell method issue

### ISSUE THREAD

**Issue by JosiahPraeman** [nus-cs2030/2324-s2#303]
# Summary

Recursion error

## Description:

The method that i used for my jshell is delivering the correct outputs but in the reverse order......

![Screenshot 2024-03-23 150617](https://github.com/nus-cs2030/2324-s2/assets/111676024/4934b6fb-fd37-4147-81e2-a3aafd11f45d)

**Comment by ZHD1987E** [comment]
Hmm...

Did you see the #293 Q&A?

If no, the trick here is to imagine what is the next log object will appear.

**Comment by ArkAzrael** [comment]
Repost from another issue that asked the same thing.

Easiest way would just be to create an overloaded constructor for Log that takes in a value and a string. Then for f(int n) you just make a new method that takes in f(int n, string s) where s is the accumulated string. Then you just keep concetanating the string and passing it to the next f() until you hit n==1. Then you just concentanate 1 to the string and return new Log(1, accumulatedString)

TLDR return the toString() of only the final Log, not one by one

**Comment by Yiting55** [comment]
it has the same logic as the code given for the Collatz conjecture: set a base case at n==1, which is the Log object that would be added at the end. And then for the 2 conditions, instead of returning f(n/2) and f(3*n+1), you can use flatMap to do the recursion. Your order is reversed maybe because you call f(...).flatMap(...). which just needs to be reordered as ....flatMap(f(...)) so the next function is added at the end instead of the beginning.
